test/gui/blueprint.lua

```diff
@@ -202,6 +193,44 @@ end
 
 --auto enter and leave cursor-supporting mode
 
+function test.restore_mode()
+    guidm.enterSidebarMode(df.ui_sidebar_mode.Stockpiles)
+    load_ui()
+    expect.eq(df.ui_sidebar_mode.LookAround, df.global.ui.main.mode)
+    send_keys('LEAVESCREEN') -- cancel out of ui
+    expect.eq(df.ui_sidebar_mode.Stockpiles, df.global.ui.main.mode)
+    send_keys('LEAVESCREEN') -- get back to Default mode
+end
+
+function test.restore_default_on_unsupported_mode()
+    guidm.enterSidebarMode(df.ui_sidebar_mode.Default)
+    send_keys('D_BURROWS')
+    load_ui()
+    expect.eq(df.ui_sidebar_mode.LookAround, df.global.ui.main.mode)
+    send_keys('LEAVESCREEN') -- cancel out of ui
+    expect.eq(df.ui_sidebar_mode.Default, df.global.ui.main.mode)
+end
+
+function test.fail_to_find_default_mode()
+    df.global.ui.main.mode = df.ui_sidebar_mode.Burrows
```

Comment:
One more possible case where D_BURROWS could be used (I can handle this)

Reply:
D'oh I missed this one. Thanks for fixing!